src/dir.rs

```diff
@@ -67,6 +67,44 @@ impl Dir {
     pub fn iter(&mut self) -> Iter {
         Iter(self)
     }
+
+
+    /// Set the position of the directory stream, see `seekdir(3)`.
+    #[cfg(not(target_os = "android"))]
+    pub fn seek(&mut self, loc: SeekLoc) {
+        unsafe { libc::seekdir(self.0.as_ptr(), loc.0 as libc::c_long) }
+    }
+
+
+    /// Reset directory stream, see `rewinddir(3)`.
+    pub fn rewind(&mut self) {
+        unsafe { libc::rewinddir(self.0.as_ptr()) }
+    }
+
+    /// Get the current position in the directory stream.
+    ///
+    /// If this location is given to `Dir::seek`, the entries up to the previously returned
+    /// will be omitted and the iteration will start from the currently pending directory entry.
+    #[cfg(not(target_os = "android"))]
+    pub fn tell(&self) -> SeekLoc {
+        let loc = unsafe { libc::telldir(self.0.as_ptr()) };
+        SeekLoc(loc.into())
+    }
+}
+
+#[cfg(not(target_os = "android"))]
+#[derive(Clone, Copy, Debug)]
+pub struct SeekLoc(i64);
+
+#[cfg(not(target_os = "android"))]
+impl SeekLoc {
+    pub unsafe fn from_raw(loc: i64) -> Self {
+        SeekLoc(loc)
+    }
+
+    pub fn to_raw(&self) -> i64 {
```

Comment:
In WASI we need to accept/return the cookie in a C-compatible way, so we need to be able access the raw cookie.

Creating a SeekLoc from an integer is obviously unsafe, hence marked as such.

Reply:
Does the client need to do anything with the returned cookie?  Or just hold onto it for awhile, and then return it to Nix?  If the latter, then you shouldn't need `from_raw` and `to_raw`.  If you need to pass it to C code, then you might want to add `#[repr(transparent)]` to the struct definition.

Reply:
In WASI, in the readdir-like syscall we receive the cookie as a 64-bit integer. Using `#[repr(transparent)]` would work, but would require `std::mem::transmute`, which we'd like to avoid - the whole point of using nix is to avoid as much unsafe code as possible.